pkg/migration-tool/utils/utils.go

```diff
@@ -0,0 +1,94 @@
+package utils
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/prompb"
+)
+
+const (
+	Megabyte = 1024 * 1024
+	LabelJob = "job"
+)
+
+// CreateReadClient creates a new read client that can be used to fetch promb samples.
+func CreateReadClient(name, urlString string, readTimeout model.Duration) (*Client, error) {
+	parsedUrl, err := url.Parse(urlString)
+	if err != nil {
+		return nil, fmt.Errorf("parsing-url: %w", err)
+	}
+	readClient, err := NewClient(name, "read", &clientConfig{
```

Comment:
why don't we change NewClient to have signature NewClient(remoteName, clientType, urlString, timeout) that way we can avoid having CreateReadClient/CreateWriteClient because NewClient will be easy to use. It's a bit weird to have CreateReadClient return a client that can both read and write.

Reply:
Sure. Updated and hence resolving